includes/Core/Modules/REST_Modules_Controller.php

```diff
@@ -435,9 +436,28 @@ private function get_rest_routes() {
 							if ( ! $module instanceof Module_With_Settings ) {
 								return new WP_Error( 'invalid_module_slug', __( 'Module does not support settings.', 'google-site-kit' ), array( 'status' => 400 ) );
 							}
-							return new WP_REST_Response( $module->get_settings()->get() );
+
+							if ( $can_manage_options() ) {
+								return new WP_REST_Response( $module->get_settings()->get() );
+							}
+
+							$settings = $module->get_settings();
+							if ( $settings instanceof Setting_With_ViewOnly_Keys_Interface ) {
+								$view_only_keys = $settings->get_view_only_keys();
+								$settings       = $settings->get();
+
+								if ( empty( $view_only_keys ) || empty( $settings ) ) {
+									return array();
+								}
+
+								$view_only_settings = array_intersect_key( $settings, array_flip( $view_only_keys ) );
+
+								return new WP_REST_Response( $view_only_settings );
+							}
+
+							return new WP_Error( 'no_view_only_settings', __( 'No view-only settings available.', 'google-site-kit' ) );
```

Comment:
The translatable message shouldn't be necessary here, this is mostly an internal error that we don't expect to encounter, except maybe in tests. Adding this adds a new string to be translated for the plugin, but it shouldn't be possible to trigger, but also, this message probably wouldn't be very useful to the user to see either.